assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -26,63 +26,63 @@ import invariant from 'invariant';
  */
 import API from 'googlesitekit-api';
 import Data from 'googlesitekit-data';
-import { isValidPropertyID, parsePropertyID, isValidProfileName } from '../util';
+import { isValidPropertyID, isValidProfileName, isValidAccountID } from '../util';
 import { STORE_NAME, PROFILE_CREATE } from './constants';
 import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
 const { createRegistrySelector } = Data;
 
 const fetchGetProfilesStore = createFetchStore( {
 	baseName: 'getProfiles',
-	controlCallback: ( { propertyID } ) => {
-		const { accountID } = parsePropertyID( propertyID );
+	controlCallback: ( { accountID, propertyID } ) => {
 		return API.get( 'modules', 'analytics', 'profiles', {
 			accountID,
 			propertyID,
 		}, {
 			useCache: false,
 		} );
 	},
-	reducerCallback: ( state, profiles, { propertyID } ) => {
+	reducerCallback: ( state, profiles, { accountID, propertyID } ) => {
 		return {
 			...state,
 			profiles: {
 				...state.profiles,
-				[ propertyID ]: [ ...profiles ],
+				[ `${ accountID }::${ propertyID }` ]: [ ...profiles ],
 			},
 		};
 	},
-	argsToParams: ( propertyID ) => {
+	argsToParams: ( accountID, propertyID ) => {
+		invariant( isValidAccountID( accountID ), 'a valid account ID is required to fetch profiles for.' );
 		invariant( isValidPropertyID( propertyID ), 'a valid property ID is required to fetch profiles for.' );
-		return { propertyID };
+		return { accountID, propertyID };
 	},
 } );
 
 const fetchCreateProfileStore = createFetchStore( {
 	baseName: 'createProfile',
-	controlCallback: ( { propertyID, profileName } ) => {
-		const { accountID } = parsePropertyID( propertyID );
+	controlCallback: ( { accountID, propertyID, profileName } ) => {
 		return API.set( 'modules', 'analytics', 'create-profile', {
 			accountID,
 			propertyID,
 			profileName,
 		} );
 	},
-	reducerCallback: ( state, profile, { propertyID } ) => {
+	reducerCallback: ( state, profile, { accountID, propertyID } ) => {
 		return {
 			...state,
 			profiles: {
 				...state.profiles,
-				[ propertyID ]: [
+				[ `${ accountID }::${ propertyID }` ]: [
 					...( state.profiles[ propertyID ] || [] ),
```

Comment:
The key needs to be the same here as the one on the line above where it includes both IDs (i.e. `${ accountID }::${ propertyID }`).

Reply:
@aaemnnosttv this seems to be correct now. Can you confirm?

Reply:
The key on L74 is correct and the key on L75 should match 🙂